lib/std/fs/test.zig

```diff
@@ -73,6 +73,38 @@ test "directory operations on files" {
     file.close();
 }
 
+test "file operations on directories" {
+    var tmp_dir = tmpDir(.{});
+    defer tmp_dir.cleanup();
+
+    const test_dir_name = "test_dir";
+
+    try tmp_dir.dir.makeDir(test_dir_name);
+
+    testing.expectError(error.IsDir, tmp_dir.dir.createFile(test_dir_name, .{}));
+    testing.expectError(error.IsDir, tmp_dir.dir.deleteFile(test_dir_name));
+    testing.expectError(error.IsDir, tmp_dir.dir.readFileAlloc(testing.allocator, test_dir_name, std.math.maxInt(usize)));
+    // note: the `.write = true` is necessary to ensure the error occurs on all platforms
+    testing.expectError(error.IsDir, tmp_dir.dir.openFile(test_dir_name, .{ .write = true }));
```

Comment:
I was unsure about how to handle the differing errors per-platform for the read-only case here (as mentioned [here](https://github.com/ziglang/zig/pull/5648#issuecomment-647072763):

> On Linux, when opening a file with read-only permissions, there is no way to determine whether the file descriptor is a directory handle without an additional syscall.

That is, `open` on Linux only returns `EISDIR` when "[pathname refers to a directory and the access requested involved writing](https://man7.org/linux/man-pages/man2/open.2.html)".

Seems like it'd be good to test the read-only case, but I'm not sure if that should involve changing the implementation of `openFile` to behave consistently across platforms (i.e. do a stat on Linux and return `error.IsDir`) or if the test should have a per-platform implementation. This might be worth opening a new issue for, though.